pkg/kubelet/dockershim/network/hostport/hostport_manager.go

```diff
@@ -231,6 +238,11 @@ func (hm *hostportManager) Remove(id string, podPortMapping *PodPortMapping) (er
 		}
 	}
 
+	// exit if there is nothing to remove
+	if len(existingChainsToRemove) == 0 {
+		return nil
```

Comment:
Not familiar with this code, but is there any chance that exiting early, which prevents calling `closeHostPorts`, could cause a leak if the previous run crashed after `syncIPTables` and before `closeHostPorts`?

Reply:
hmm you are absolutely right

Reply:
however, this state is weird, so you still have hostportMappings but you don´t have iptables rules :thinking: 
I´m trying to remember why we added this, it was for dual stack for sure, because we called twice Remove()

Reply:
I think we can return closeHostports() here too, the method goes through the map so it doesn´t seem risky.
Holding iptables lock for nothing in a busy environment is expensive ....